File: src/main/java/org/jcodec/containers/mp4/boxes/MP4ABox.java
package org.jcodec.containers.mp4.boxes;

import js.nio.ByteBuffer;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * @author The JCodec project
 * 
 */
public class MP4ABox extends Box {

    public MP4ABox(Header header) {
        super(header);
    }

    private int val;

    protected void doWrite(ByteBuffer out) {
        out.putInt(val);
    }

    public void parse(ByteBuffer input) {
        val = input.getInt();
    }

    @Override
    public int estimateSize() {
        return 12;
    }
}

File: src/main/java/org/jcodec/containers/mp4/boxes/MP4ABox.kt
package org.jcodec.containers.mp4.boxes

import java.nio.ByteBuffer

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * @author The JCodec project
 */
class MP4ABox(header: Header) : Box(header) {
    private var `val` = 0
    override fun doWrite(out: ByteBuffer) {
        out.putInt(`val`)
    }

    override fun parse(buf: ByteBuffer) {
        `val` = buf.int
    }

    override fun estimateSize(): Int {
        return 12
    }
}